### PR TITLE
Adding logic for actions

### PR DESCRIPTION
**Pull request by brawner**
This adds support for actions to rqt_py_common. This is necessary to port rqt_action

It depends on https://github.com/ros2/rosidl_python/pull/21

### CODE REVIEW

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (mlautman) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:46]: why only filter prefixes for actions?

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (mlautman) [inline rqt_py_common/src/rqt_py_common/message_helpers.py]: I'm not a fan of the two for loops. Why not use `zip`?

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (mlautman) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:43]: fancy

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (mlautman) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:158]: Nice!

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (mlautman) [inline rqt_py_common/src/rqt_py_common/message_helpers.py]: Nit: you should use the keys again

**Review by mlautman — COMMENTED** [review]
+1 after addressing feedback

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (brawner) [reply rqt_py_common/src/rqt_py_common/message_helpers.py:46]: Because the previous logic would collect both ```.msg``` files that were ```srv/*.msg``` and ```msg/*.msg```. Unfortunately there is no rosidl_interface yet that includes an ```.action``` suffix and ```.idl``` is not specific to actions.

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (brawner) [reply rqt_py_common/src/rqt_py_common/message_helpers.py]: I suppose it's a little less likely to be buggy with zip.

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (brawner) [reply rqt_py_common/src/rqt_py_common/message_helpers.py]: Done

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (dirk-thomas) [inline rqt_py_common/src/rqt_py_common/message_helpers.py:46]: The current code base doesn't have `.idl` file. Therefore this should check for the extension `.action`.

**Review by dirk-thomas — COMMENTED** [review]
Beside the comment I think this needs to wait until the generated Python API for actions has been reviewed and merged, e.g. if there is an action class being generated or not.

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (brawner) [reply rqt_py_common/src/rqt_py_common/message_helpers.py:46]: Oh strange. I guess I'm working from some older branches. I was hoping for an .action interface to be added. 

Updated

**Comment by brawner** [comment]
@dirk-thomas @mlautman This PR has been updated now that the rosidl_python PR has been merged.

File: rqt_py_common/src/rqt_py_common/message_helpers.py
Comment (dirk-thomas) [reply rqt_py_common/src/rqt_py_common/message_helpers.py:46]: There are no `.idl` files in the current master branch yet. They are still named `.action`.

Comment (brawner) [reply rqt_py_common/src/rqt_py_common/message_helpers.py:46]: When I generate the rosidl_interfaces for test_msgs, this is the list that's created and added to the rosidl_interfaces resource_index. I haven't noticed an action/Fibonacci.action being added to this list. Do you know where this happens so I can pull the correct branch and test this out with .action? I've done a basic vcs pull from master, but maybe I'm missing something.

```
action/Fibonacci.idl
action/Fibonacci_Feedback.msg
action/Fibonacci_Goal.srv
action/Fibonacci_Goal_Request.msg
action/Fibonacci_Goal_Response.msg
action/Fibonacci_Result.srv
action/Fibonacci_Result_Request.msg
action/Fibonacci_Result_Response.msg
msg/BoundedArrayNested.idl
msg/BoundedArrayNested.msg
msg/BoundedArrayPrimitives.idl
msg/BoundedArrayPrimitives.msg
msg/BoundedArrayPrimitivesNested.idl
msg/BoundedArrayPrimitivesNested.msg
msg/Builtins.idl
msg/Builtins.msg
msg/DynamicArrayNested.idl
msg/DynamicArrayNested.msg
msg/DynamicArrayPrimitives.idl
msg/DynamicArrayPrimitives.msg
msg/DynamicArrayPrimitivesNested.idl
msg/DynamicArrayPrimitivesNested.msg
msg/DynamicArrayStaticArrayPrimitivesNested.idl
msg/DynamicArrayStaticArrayPrimitivesNested.msg
msg/Empty.idl
msg/Empty.msg
msg/Nested.idl
msg/Nested.msg
msg/Primitives.idl
msg/Primitives.msg
msg/StaticArrayNested.idl
msg/StaticArrayNested.msg
msg/StaticArrayPrimitives.idl
msg/StaticArrayPrimitives.msg
msg/StaticArrayPrimitivesNested.idl
msg/StaticArrayPrimitivesNested.msg
srv/Empty.idl
srv/Empty.srv
srv/Empty_Request.msg
srv/Empty_Response.msg
srv/Primitives.idl
srv/Primitives.srv
srv/Primitives_Request.msg
srv/Primitives_Response.msg
```

**Review by mlautman — APPROVED** [review]

**Comment by dirk-thomas** [comment]
@mlautman The consensus we have is that PRs on this repo are waiting for an approval from the maintainer (in contrast to plugin repos). Please do so on future PRs. If there is no response feel free to comment requesting for feedback but don't merge them without.

**Pull request merged**
